server/src/test/java/io/crate/analyze/AlterTableDropColumnAnalyzerTest.java

```diff
@@ -0,0 +1,254 @@
+        assertThatThrownBy(() -> analyze("ALTER TABLE t2 DROP COLUMN o['oo']['ooa']"))
+            .isExactlyInstanceOf(UnsupportedOperationException.class)
+            .hasMessage("Dropping column: o['oo']['ooa'] which is part of the 'PARTITIONED BY' columns is not allowed");
+    }
+
+    @Test
+    public void test_drop_pk_column_is_not_allowed() throws Exception {
+        e = SQLExecutor.builder(clusterService)
+            .addTable(TableDefinitions.USER_TABLE_MULTI_PK_DEFINITION)
+            .addTable("CREATE TABLE t2 (o object AS(oo object AS(ooa int)), PRIMARY KEY (o['oo']['ooa']))")
+            .build();
+
+        // id is also part of clustered by, but pk is checked first
+        assertThatThrownBy(() -> analyze("ALTER TABLE users_multi_pk DROP COLUMN id"))
+            .isExactlyInstanceOf(UnsupportedOperationException.class)
+            .hasMessage("Dropping column: id which is part of the PRIMARY KEY is not allowed");
+        assertThatThrownBy(() -> analyze("ALTER TABLE users_multi_pk DROP name"))
+            .isExactlyInstanceOf(UnsupportedOperationException.class)
+            .hasMessage("Dropping column: name which is part of the PRIMARY KEY is not allowed");
+
+        assertThatThrownBy(() -> analyze("ALTER TABLE t2 DROP COLUMN o['oo']['ooa']"))
+            .isExactlyInstanceOf(UnsupportedOperationException.class)
+            .hasMessage("Dropping column: o['oo']['ooa'] which is part of the PRIMARY KEY is not allowed");
+    }
+
+    @Test
+    public void test_drop_column_used_in_a_generated_col_is_not_allowed() throws Exception {
+        e = SQLExecutor.builder(clusterService)
+            .addTable("CREATE TABLE t(a int, b generated always as (a + 1))")
+            .addTable("CREATE TABLE t1(o object AS (oo object AS(ooa int)), b generated always as (o['oo']['ooa'] + 1))")
+            .build();
+
+        assertThatThrownBy(() -> analyze("ALTER TABLE t DROP a"))
+            .isExactlyInstanceOf(UnsupportedOperationException.class)
+            .hasMessage("Dropping column: a which is used to produce values for generated column is not allowed");
+
+        assertThatThrownBy(() -> analyze("ALTER TABLE t1 DROP o['oo']['ooa']"))
+            .isExactlyInstanceOf(UnsupportedOperationException.class)
+            .hasMessage("Dropping column: o['oo']['ooa'] which is used to produce values for generated column is not allowed");
+    }
+
+    @Test
+    public void test_drop_column_used_in_a_full_text_index_is_not_allowed() throws Exception {
+        e = SQLExecutor.builder(clusterService)
+            .addTable("CREATE TABLE t(a int, b text INDEX using FULLTEXT)")
+            .build();
+
+        assertThatThrownBy(() -> analyze("ALTER TABLE t DROP b"))
+            .isExactlyInstanceOf(UnsupportedOperationException.class)
+            .hasMessage("Dropping column: b which is used in a FULLTEXT INDEX is not allowed");
+    }
```

Comment:
I think dropping column which is indexed on its own is fine.
I mean we cannot drop part of composite index.

Smth like using fulltext(title, description) = DROP title or DROP description should be restricted. Otherwise FT index would be broken?